lib/ansible/modules/notification/mail.py

```diff
@@ -219,7 +220,7 @@ def main():
         argument_spec=dict(
             username=dict(type='str'),
             password=dict(type='str', no_log=True),
-            host=dict(type='str', default='localhost'),
+            host=dict(type='str', default='127.0.0.1'),
```

Comment:
The default shouldn't be changed just because one particular platform running an old Python version happens to provide a less than helpful error message when the connection is refused. Also, defaults shouldn't be changed for bug fixes that will be backported.

The examples don't need to be switched away from localhost either.

Reply:
It's not providing a less than helpful error message, it plain fails. And there might be other reasons for it to fail (e.g. resolver issues or /etc/hosts being empty) so 127.0.0.1 from that respect is a better default.

Same for the examples, 127.0.0.1 is a better guarantee it will work. So IMO it is an improvement after what we learned.

Reply:
Where are you seeing a failure with localhost?

Reply:
I was getting them when smtpserver.py was fixed to only do non-TLS connections with the most basic integration test that was lacking a host-parameter (hence using 'localhost').

Reply:
I tried that on Ubuntu 14.04 but haven't been able to reproduce the issue you're seeing. Using localhost works for me as long as smtpserver.py is running and TLS isn't used.

Reply:
Ok, since I can't test I only have circumstancial evidence. The reports about this specific error also indicated that using an IP address would fix that issue. So maybe the server wasn't running at all (which I can't see from the async task). I am working in the dark here, and already spend too much time on corner cases I am not interested in really. I just want to avoid the next guy fixing something and breaking everything else. This PR accomplishes that, so for me it's over and done for now.

Reply:
Sorry, I must be missing some context here. What reports about localhost vs 127.0.0.1 are you referring to? The only issue I am aware of is with earlier versions of this PR where smtpserver.py wasn't running.

Reply:
StackOverflow